hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/INodeMap.java

```diff
@@ -191,8 +202,28 @@ static INodeMap newInstance(INodeDirectory rootDir,
     return map.iterator();
   }
 
-  private INodeMap(INodeDirectory rootDir, FSNamesystem ns) {
+  public Set<INode> rangeKeys() {
+    return ((PartitionedGSet)map).entryKeySet();
+  }
+
+  private static void setStaticField(Configuration conf) {
+    numSpaceKeyDepth = conf.getInt(DFSConfigKeys.DFS_NAMENODE_INOD_NAMESPACE_KEY_DEPTH,
+        DFSConfigKeys.DFS_NAMENODE_INOD_NAMESPACE_KEY_DEPTH_DEFAULT);
+    if (numSpaceKeyDepth < 1) {
+      numSpaceKeyDepth = DFSConfigKeys.DFS_NAMENODE_INOD_NAMESPACE_KEY_DEPTH_DEFAULT;
+    }
+
+    numRanges = conf.getLong(DFSConfigKeys.DFS_NAMENODE_INOD_NUM_RANGES,
```

Comment:
Can we add a check here to make sure numRanges is power of 2? thanks,

Reply:
Thanks @xinglin for the comment.
I will update it later.